### PR TITLE
Colors migration - 2d - Migrate SurveyToggle

### PR DESCRIPTION
**Pull request by damianmef**

### CODE REVIEW

File: lib/styles/settings/_survey-toggle.scss [hunk 1]
```diff
@@ -0,0 +1,54 @@
+@import '../settings/colors/tokens';
+
+$survey-toggle-size: 48px;
```
Comment (kvas-damian) [inline lib/styles/settings/_survey-toggle.scss:3]: why did you decided to extract it to a separate file?

File: lib/js/components/SurveyToggle/SurveyToggle.vue [hunk 2]
```diff
@@ -36,14 +37,62 @@
...
+@import '../../../styles/settings/survey-toggle';
+@import '../../../styles/components/survey-toggle';
 
 .surveyToggle {
 	$self: &;
 
+	@each $color-name, $color-map in $survey-toggle-colors {
+		&.-#{$color-name} {
+			#{$self}__toggle {
+				@include setSurveyToggleNormalState(
```
Comment (kvas-damian) [inline lib/js/components/SurveyToggle/SurveyToggle.vue:50]: IMO, it makes sense to seaprate mixins for `toggle` and `ring`. Parameter sets are separate for both use-cases, so there is no reason to complicate it

Comment (damianmef) [reply lib/js/components/SurveyToggle/SurveyToggle.vue:137]: Right. Refactored.

File: lib/styles/settings/_survey-toggle.scss (hunk 1 above)
Comment (damianmef) [reply lib/styles/settings/_survey-toggle.scss]: After discussion I removed all external settings for components (except Button).

File: lib/js/components/SurveyToggle/SurveyToggle.vue [hunk 3]
```diff
@@ -35,14 +36,139 @@
...
+		'icon': $color-default-text-inverted,
+		'background-hovered': $color-neutral-background-strong-hovered,
+		'ripple': $color-default-ripple-inverted,
+		'disabled': (
+			'background': $color-neutral-background-strong-disabled,
+		),
+	),
+);
+
+@mixin setSurveyContentOtherStates($disabled-text) {
```
Comment (kvas-damian) [inline lib/js/components/SurveyToggle/SurveyToggle.vue:94]: does makes sense to call this method `...OtherStates` now?

File: lib/js/components/SurveyToggle/SurveyToggle.vue [hunk 4]
```diff
@@ -35,14 +36,139 @@
...
+		&.-#{$color-name} {
+			#{$self}__toggle {
+				@include setSurveyToggleNormalState(
+					map-get($color-map, 'background'),
+					map-get($color-map, 'color'),
+					map-get($color-map, 'ripple')
+				);
+			}
+			#{$self}__ring {
+				@include setSurveyRingNormalState(map-get($color-map, 'border'));
```
Comment (kvas-damian) [inline lib/js/components/SurveyToggle/SurveyToggle.vue:145]: does it makes sense to introduce mixi for one attribute?

**Review by kvas-damian — APPROVED** [review]

File: lib/js/components/SurveyToggle/SurveyToggle.vue [hunk 5]
```diff
@@ -35,14 +36,124 @@
...
 
 $survey-toggle-size: 48px;
 
-$color-disabled-neutral: #000; // IT WILL BE REPLACED IN NEXT PR
+$survey-toggle-colors: (
+	'primary': (
+		'background': $color-primary-background-weak,
+		'border': $color-primary-border,
+		'color': $color-primary-text,
+		'icon': $color-default-icon-inverted,
```
[the remark this answers is not in the document]
Comment (damianmef) [reply lib/js/components/SurveyToggle/SurveyToggle.vue:48]: Thanks, It's fixed.

[the remark this answers is not in the document]
Comment (kvas-damian) [reply lib/js/components/SurveyToggle/SurveyToggle.vue:48]: @damianmef it isn't fixed. You still define `icon` color for `primary` and use it for all color schemes

**Pull request merged**